File: project2/Organism.java
abstract class Organism {
    int strength, initiative, age;
    int x, y;
    World world;
    boolean alive = true;
    
    public Organism(int str, int init, int x, int y, World w) {
        this.strength = str;
        this.initiative = init;
        this.x = x;
        this.y = y;
        this.world = w;
        this.age = 0;
    }
    
    public abstract void action();
    public abstract void collision(Organism other);
    public abstract char draw();
    
    public boolean isAlive() {
        return alive;
    }
}

File: project3/organism.py
from abc import ABC, abstractmethod
import pickle

class Organism(ABC):
    def __init__(self, strength, initiative, x, y, world):
        self.strength = strength
        self.initiative = initiative 
        self.x = x
        self.y = y
        self.world = world
        self.age = 0
        self.alive = True
    
    @abstractmethod
    def action(self):
        pass
    
    @abstractmethod
    def collision(self, other):
        pass
    
    @abstractmethod
    def draw(self):
        pass
    
    def is_alive(self):
        return self.alive
